File: samples/CleanArchitectureSample/src/Products.Module/Domain/Product.cs
using System.Text.Json.Serialization;

namespace Products.Module.Domain;

/// <summary>
/// Represents a product in the catalog.
/// This is a domain entity - the core business object for the Products bounded context.
/// </summary>
public record Product(
    string Id,
    string Name,
    string Description,
    decimal Price,
    int StockQuantity,
    ProductStatus Status,
    DateTime CreatedAt,
    DateTime? UpdatedAt = null);

/// <summary>
/// Represents the availability status of a product.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ProductStatus
{
    Draft,
    Active,
    OutOfStock,
    Discontinued
}

File: samples/CleanArchitectureSample/src/Web/src/lib/types/product.ts
export type ProductStatus = 'Draft' | 'Active' | 'OutOfStock' | 'Discontinued';

export interface Product {
  id: string;
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
  status: ProductStatus;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateProductRequest {
  name: string;
  description: string;
  price: number;
  stockQuantity?: number;
}

export interface UpdateProductRequest {
  name?: string;
  description?: string;
  price?: number;
  stockQuantity?: number;
  status?: ProductStatus;
}

export const PRODUCT_STATUS_COLORS: Record<ProductStatus, string> = {
  Draft: 'bg-gray-100 text-gray-800',
  Active: 'bg-green-100 text-green-800',
  OutOfStock: 'bg-red-100 text-red-800',
  Discontinued: 'bg-orange-100 text-orange-800'
};
